### PR TITLE
Passing optional arguments to Comunica

### PR DESCRIPTION
**Pull request by danielbeeke**
See https://github.com/LDflex/LDflex/issues/70

### CODE REVIEW

File: src/ComunicaEngine.js [hunk 1]
```diff
@@ -10,8 +10,10 @@ export default class ComunicaEngine {
    * The default source can be a single URL, an RDF/JS Datasource,
    * or an array with any of these.
    */
-  constructor(defaultSource) {
+  constructor(defaultSource, optionalArguments) {
```
Comment (RubenVerborgh) [inline src/ComunicaEngine.js:13]:
```suggestion
  constructor(defaultSource, options) {
```

File: src/ComunicaEngine.js [hunk 2]
```diff
@@ -10,8 +10,10 @@ export default class ComunicaEngine {
...
     this._engine = DefaultEngine;
+    this._optionalArguments = optionalArguments;
```
Comment (RubenVerborgh) [inline src/ComunicaEngine.js:15]:
```suggestion
    this._options = options;
```

File: src/ComunicaEngine.js [hunk 3]
```diff
@@ -28,7 +30,7 @@ export default class ComunicaEngine {
     const sources = await (source ? this.parseSources(source) : this._sources);
     if (sources.length !== 0) {
       // Execute the query and yield the results
-      const queryResult = await this._engine.query(sparql, { sources });
+      const queryResult = await this._engine.query(sparql, Object.assign({ sources }, this._optionalArguments));
```
Comment (RubenVerborgh) [inline src/ComunicaEngine.js:33]:
```suggestion
      const queryResult = await this._engine.query(sparql, { sources, ...ths._options });
```

**Review by RubenVerborgh — CHANGES_REQUESTED** [review]

**Comment by danielbeeke** [comment]
Interesting enough, the options are passed. But I can not get it work. 
I am using it to pass a httpProxyHandler like:

```
        const queryEngine = new ComunicaEngine(uri, {
          'httpProxyHandler': this.form.proxy
        });

        const path = new PathFactory({ context: this.pathContext, queryEngine });
        const object = path.create({ subject: namedNode(uri) })
```

**Comment by danielbeeke** [comment]
Can be closed in favor of #24 24

**Pull request closed without merging**